python-backend/input.py

```diff
@@ -18,4 +21,13 @@ def convert(polynomial: str) -> str:
     :param polynomial: incoming polynomial entered by user in web frontend
     :return: python parse-able polynomial
     """
-    return re.sub(r'([0-9])+([a-zA-Z])', '\\1*\\2', polynomial.replace('^', '**'))
+    expression = re.sub(r'([0-9]+)([a-zA-Z])', '\\1*\\2',
+                        polynomial.replace('^', '**').replace(' ', '').replace(')(', ')*('))
+    expression = re.sub(r'([0-9a-zA-Z])(\()', '\\1*\\2', expression)
+    logger.debug(f"input: {polynomial} output: {expression}")
+    return expression
+
+
+class Expression(BaseModel):
```

Comment:
Will this class be extended in the future? If not, is a single string sufficient (rather than a class)?

Reply:
It will be extended.